src/Common/src/System/Collections/Generic/LowLevelList.cs

```diff
@@ -1,606 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-/*============================================================
-**
-**
-** Private version of List<T> for internal System.Private.CoreLib use. This
-** permits sharing more source between BCL and System.Private.CoreLib (as well as the
-** fact that List<T> is just a useful class in general.)
-**
-** This does not strive to implement the full api surface area
-** (but any portion it does implement should match the real List<T>'s
-** behavior.)
-**
-** This file is a subset of System.Collections\System\Collections\Generics\List.cs
-** and should be kept in sync with that file.
-** 
-===========================================================*/
-
-using System;
-using System.Diagnostics;
-using System.Diagnostics.Contracts;
-
-namespace System.Collections.Generic
-{
-    // Implements a variable-size List that uses an array of objects to store the
-    // elements. A List has a capacity, which is the allocated length
-    // of the internal array. As elements are added to a List, the capacity
-    // of the List is automatically increased as required by reallocating the
-    // internal array.
-    // 
-    /// <summary>
-    /// LowLevelList with no interface implementation to minimize both code and data size
-    /// Data size is smaller because there will be minimal virtual function table.
-    /// Code size is smaller because only functions called will be in the binary.
-    /// Use LowLevelListWithIList<T> for IList support
-    /// </summary>
-    [DebuggerDisplay("Count = {Count}")]
-#if TYPE_LOADER_IMPLEMENTATION
-    [System.Runtime.CompilerServices.ForceDictionaryLookups]
```

Comment:
Type loader uses this to suppress recursive generic lookups from within it's codebase. I'm not sure we have a replacement if we get a normal List instead.

Cc @davidwrighton

Reply:
Another feature/benefit of LowLevelList is that it is the instantiations have significantly smaller binary footprint than List. It would be useful to measure the AOT binary size impact of changes like these on ProjectN apps and ProjectN shared library.

I think we will want to keep LowLevelList, in some places at least - in particular where the code is not shared with CoreCLR, for both these reasons.

LowLevelListWithIList should be ok to remove.